src/test/java/GmailActionableEmailsServletTest.java

```diff
@@ -0,0 +1,166 @@
+        .thenReturn(SUBJECT_LINE_PHRASES_STRING);
+    Mockito.when(request.getParameter("unreadOnly")).thenReturn(String.valueOf(true));
+
+    servlet.doGet(request, response);
+    Assert.assertEquals(400, response.getStatus());
+  }
+
+  @Test
+  public void subjectLinePhrasesEmpty() throws Exception {
+    Mockito.when(request.getParameter("subjectLinePhrases")).thenReturn("");
+    Mockito.when(request.getParameter("unreadOnly")).thenReturn(String.valueOf(true));
+    Mockito.when(request.getParameter("nDays")).thenReturn(String.valueOf(DEFAULT_N_DAYS));
+
+    servlet.doGet(request, response);
+    Assert.assertEquals(400, response.getStatus());
+  }
+
+  @Test
+  public void unreadOnlyEmpty() throws Exception {
+    Mockito.when(request.getParameter("subjectLinePhrases"))
+        .thenReturn(SUBJECT_LINE_PHRASES_STRING);
+    Mockito.when(request.getParameter("unreadOnly")).thenReturn("");
+    Mockito.when(request.getParameter("nDays")).thenReturn(String.valueOf(DEFAULT_N_DAYS));
+
+    servlet.doGet(request, response);
+    Assert.assertEquals(400, response.getStatus());
+  }
+
+  @Test
+  public void nDaysEmpty() throws Exception {
+    Mockito.when(request.getParameter("subjectLinePhrases"))
+        .thenReturn(SUBJECT_LINE_PHRASES_STRING);
+    Mockito.when(request.getParameter("unreadOnly")).thenReturn(String.valueOf(true));
+    Mockito.when(request.getParameter("nDays")).thenReturn("");
+
+    servlet.doGet(request, response);
+    Assert.assertEquals(400, response.getStatus());
+  }
+
+  @Test
+  public void nDaysNegative() throws Exception {
+    Mockito.when(request.getParameter("subjectLinePhrases"))
+        .thenReturn(SUBJECT_LINE_PHRASES_STRING);
+    Mockito.when(request.getParameter("unreadOnly")).thenReturn(String.valueOf(true));
+    Mockito.when(request.getParameter("nDays")).thenReturn(String.valueOf(NEGATIVE_N_DAYS));
+
+    servlet.doGet(request, response);
+    Assert.assertEquals(400, response.getStatus());
+  }
+
+  @Test
+  public void nDaysZero() throws Exception {
+    Mockito.when(request.getParameter("subjectLinePhrases"))
+        .thenReturn(SUBJECT_LINE_PHRASES_STRING);
+    Mockito.when(request.getParameter("unreadOnly")).thenReturn(String.valueOf(true));
+    Mockito.when(request.getParameter("nDays")).thenReturn(String.valueOf(0));
+
+    servlet.doGet(request, response);
+    Assert.assertEquals(400, response.getStatus());
+  }
+
+  @Test
+  public void validResponse() throws Exception {
+    Mockito.when(request.getParameter("subjectLinePhrases"))
+        .thenReturn(SUBJECT_LINE_PHRASES_STRING);
+    Mockito.when(request.getParameter("unreadOnly")).thenReturn(String.valueOf(true));
+    Mockito.when(request.getParameter("nDays")).thenReturn(String.valueOf(DEFAULT_N_DAYS));
+
+    // List passed to getActionableEmails should be words from query string
+    // parsed by comma separator
+    List<String> listOfSubjectWords = Arrays.asList(SUBJECT_LINE_PHRASES_STRING.split(","));
```

Comment:
I suggest defining a separate constant for the expected list. That way you are testing just the inputs/outputs instead of duplicating the parsing logic in the test.

Reply:
Sounds good to me!